lib/jormungandr/src/Cardano/Wallet/Jormungandr.hs

```diff
@@ -116,27 +120,48 @@ serveWallet (cfg, sb, tr) databaseDir listen lj beforeMainLoop = do
             let nPort = Port $ baseUrlPort $ _restApi cp
             waitForService "Jörmungandr" (sb, tr) nPort $
                 waitForNetwork nl defaultRetryPolicy
-            apiLayer tr nl >>= startServer tr nPort nl
+            apiLayer tr (toWLBlock <$> nl) >>= startServer tr nPort nl
             pure ExitSuccess
         Left e -> handleNetworkStartupError e
   where
     startServer
         :: Trace IO Text
         -> Port "node"
-        -> NetworkLayer IO t (Block Tx)
+        -> NetworkLayer IO t Binary.Block
         -> ApiLayer s t k
         -> IO ()
     startServer tracer nPort nl wallet = do
         let (_, bp) = staticBlockchainParameters nl
+        spl <- SP.newStakePoolLayer (toSPBlock <$> nl) tracer
         Server.withListeningSocket listen $ \(wPort, socket) -> do
             let tracerIPC = appendName "daedalus-ipc" tracer
             let tracerApi = appendName "api" tracer
             let settings = Warp.defaultSettings
                     & setBeforeMainLoop (beforeMainLoop (Port wPort) nPort bp)
             let ipcServer = daedalusIPC tracerIPC wPort
-            let apiServer = Server.start settings tracerApi socket wallet
+            let apiServer = Server.start settings tracerApi socket wallet spl
             race_ ipcServer apiServer
 
+    -- Will be used to connect "Cardano.Wallet.StakePool.Metrics" with
+    -- our networkLayer.
+    toSPBlock :: Binary.Block -> (BlockHeader, PoolId)
+    toSPBlock b =
+        (convertHeader header, toJust $ Binary.producedBy header)
+      where
+        header = Binary.header b
+        convertHeader :: Binary.BlockHeader -> BlockHeader
+        convertHeader h = BlockHeader
+            (Binary.slot h)
+            (Quantity $ fromIntegral $ Binary.chainLength h)
+            (Binary.parentHeaderHash h)
+        toJust (Just x) = x
+        toJust Nothing = error
+            "Expected blockheader to contain the id of the\
+            \producing pool, which only happens on Praos/Genesis-blockchains\
+            \for blocks after block0."
```

Comment:
This is still quite unsafe here. So, fetching the genesis block will throw hard in the stake pool network layer. That's bad, because we might actually fetch it more than once in the chain following algorithm! I'd stick with a `Maybe PoolId` at this stage, and do the conversion from `Maybe PoolID` to `PoolId` later ... and safely.